packages/clerk-js/src/utils/url.ts

```diff
@@ -280,18 +280,20 @@ export function setSearchParameterInHash({
 export function removeSearchParameterFromHash({
   hash = window.location.hash,
   paramName,
+  removeTrailingSlash = false,
 }: {
   hash?: string;
   paramName: string;
+  removeTrailingSlash?: boolean;
 }) {
   const h = hash.startsWith('#') ? hash.substring(1) : hash;
   const dummyUrlForHash = new URL(h, DUMMY_URL_BASE);
   dummyUrlForHash.searchParams.delete(paramName);
 
-  // The following line will prepend the hash with a `/`.
+  // The following line will prepend the hash with a `/` if removeTrailingSlash is false(default).
   // This is required for ClerkJS Components Hash router to work as expected
   // as it treats the hash as sub-path with its nested querystring parameters.
-  return dummyUrlForHash.href.replace(DUMMY_URL_BASE, '');
+  return dummyUrlForHash.href.replace(`${DUMMY_URL_BASE}${removeTrailingSlash ? '/' : ''}`, '');
```

Comment:
❓ Isn't this a little confusing? The variable name is `removeTrailingSlash`, but when it is truthy we add a trailing slash.

Reply:
I thought I removed this. My bad. This is not a requirement any more.
For the record though, we add the trailing slash to the string to be replaced by `''`, so it's essentially removed.

Reply:
This function will continue as it was before.